azuread/resource_application.go

```diff
@@ -120,15 +125,28 @@ func resourceApplicationCreate(d *schema.ResourceData, meta interface{}) error {
 
 	name := d.Get("name").(string)
 
+	appType := d.Get("type")
+
+	if appType == "native" {
+		if _, ok := d.GetOk("identifier_uris"); ok {
+			return fmt.Errorf("identifier_uris is not required for a native application")
+		}
+	} else if appType != "webapp/api" {
+		return fmt.Errorf("Error creating Azure AD Application with name %q: Unknow application type %v. Supported types are [webapp/api, native]", name, appType)
+	}
+
 	properties := graphrbac.ApplicationCreateParameters{
 		DisplayName:             &name,
-		Homepage:                expandADApplicationHomepage(d, name),
 		IdentifierUris:          tf.ExpandStringArrayPtr(d.Get("identifier_uris").([]interface{})),
 		ReplyUrls:               tf.ExpandStringArrayPtr(d.Get("reply_urls").([]interface{})),
 		AvailableToOtherTenants: p.Bool(d.Get("available_to_other_tenants").(bool)),
 		RequiredResourceAccess:  expandADApplicationRequiredResourceAccess(d),
 	}
 
+	if _, ok := d.GetOk("homepage"); ok {
+		properties.Homepage = expandADApplicationHomepage(d, name)
```

Comment:
I meant more could we replace this with:

```go
	if v, ok := d.GetOk("homepage"); ok {
		properties.Homepage = p.String(v.(string))
	}
```

Reply:
Yeah, so if we are not preserving original autosetting homepage for webapis, we can ditch it completely and replace with your suggestion. Otherwise, it would still be useful, in order to autoset homepage. I am thinking it will look something like
```
homePage := expandADApplicationHomepage(d, name)
	if appType == "native" {

		if _, ok := d.GetOk("identifier_uris"); ok {
			return fmt.Errorf("identifier_uris is not required for a native application")
		}
		// if homepage wasn't set, follow cli behavior and do not automatically set it
		if _, ok := d.GetOk("homepage"); !ok {
			homePage = nil
		}
	} else if appType != "webapp/api" {
		return fmt.Errorf("Error creating Azure AD Application with name %q: Unknow application type %v. Supported types are [webapp/api, native]", name, appType)
	}

	properties := graphrbac.ApplicationCreateParameters{
		DisplayName:             &name,
		Homepage:                homePage,
		IdentifierUris:          tf.ExpandStringArrayPtr(d.Get("identifier_uris").([]interface{})),
		ReplyUrls:               tf.ExpandStringArrayPtr(d.Get("reply_urls").([]interface{})),
		AvailableToOtherTenants: p.Bool(d.Get("available_to_other_tenants").(bool)),
		RequiredResourceAccess:  expandADApplicationRequiredResourceAccess(d),
	}
```
I am happy to go with either. It looks a bit messier if we preserve it, but it does seem like a more backwards compatible solution.

Reply:
@katbyte what are your thoughts on how we should proceed?

Reply:
Going to take a look at this todayand try and get it merged

Reply:
preserved the functionality of automatically setting a homepage when the type is webapp